interfaces/kmod/backend.go

```diff
@@ -64,18 +64,17 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementOptions, repo *interfaces.Repository) error {
 	snapName := snapInfo.Name()
 	// Get the snippets that apply to this snap
-	snippets, err := repo.SecuritySnippetsForSnap(snapInfo.Name(), interfaces.SecurityKMod)
+	spec, err := repo.SnapSpecification(b.Name(), snapName)
 	if err != nil {
-		return fmt.Errorf("cannot obtain kmod security snippets for snap %q: %s", snapName, err)
+		return fmt.Errorf("cannot obtain kmod specification for snap %q: %s", snapName, err)
 	}
 
-	// Get the files that this snap should have
-	glob := interfaces.SecurityTagGlob(snapName)
-	content, modules, err := b.combineSnippets(snapInfo, snippets)
-	if err != nil {
-		return fmt.Errorf("cannot obtain expected security files for snap %q: %s", snapName, err)
+	if len(spec.(*Specification).Modules) == 0 {
+		return nil
```

Comment:
No, this part is wrong. This should go to the `deriveContent` call. You absolutely have to call `EnsureDirState` below as that is what really applies the changes to the system. You also need to act on the things it returns.

If you move this to to `deriveContent` then you save useless construction of one map. That what what I was suggesting earlier (since most interfaces don't need kmod)